unittests/kernelcache.py

```diff
@@ -0,0 +1,191 @@
+    earthStnSpk_md5 = "a37d8d5e3023f0df7ead0e6b40d6a5b6"
+    earthHighPerPck_url = "https://pds-rings.seti.org/testrunner_support/cspyce-unit-test-kernels/earth_031228_231229_predict.bpc"
+    earthHighPerPck_md5 = "affa1da5adeee5ea4b0d7da54e4b69d7"
+# =============================================================================
+#     phobosDsk_url = "https://raw.githubusercontent.com/AndrewAnnex/SpiceyPyTestKernels/main/phobos_lores.bds"
+#     phobosDsk_md5 = "68261460433bfc67b9e57bb57f79c5c9"
+#     marsSpk_url = "https://raw.githubusercontent.com/AndrewAnnex/SpiceyPyTestKernels/main/mar022-1.bsp"
+#     marsSpk_md5 = "d8d742db3f9502571fb5a5f8c55e8e62"
+#     mroFk_url = "https://raw.githubusercontent.com/AndrewAnnex/SpiceyPyTestKernels/main/mro_v15.tf"
+#     mroFk_md5 = "a938c271be63e0e5aa2ec86db89af109"
+#     geophysical_url = "https://raw.githubusercontent.com/AndrewAnnex/SpiceyPyTestKernels/main/geophysical.ker"
+#     geophysical_md5 = "9a565ded819a9f0c6423b46f04e000db"
+#     mro2007sub_url = "https://raw.githubusercontent.com/AndrewAnnex/SpiceyPyTestKernels/main/mro_psp4_ssd_mro95a_sub.bsp"
+#     mro2007sub_md5 = "8ed34eb77b21ac611f4680806677edfb"
+#     spk430sub_url = "https://raw.githubusercontent.com/AndrewAnnex/SpiceyPyTestKernels/main/de430sub.bsp"
+#     spk430sub_md5 = "0b49545fa316f9053f5cfbcce155becc"
+# =============================================================================
+    vexboomck_url = "https://pds-rings.seti.org/testrunner_support/cspyce-unit-test-kernels/VEX_BOOM_V01.BC"
+    vexboomck_md5 = "2f4dba65649246d72836fb3b53823c3d"
+    v02swuck_url = "https://pds-rings.seti.org/testrunner_support/cspyce-unit-test-kernels/vo2_swu_ck2.bc"
+    v02swuck_md5 = "f59ef0556dfc63b55465e152f2d6f5a4"
+    cklpfkrn_url = "https://pds-rings.seti.org/testrunner_support/cspyce-unit-test-kernels/cklpfkernel.bc"
+    voyagerSclk = get_path_from_url(voyagerSclk_url)
+    earthTopoTf = get_path_from_url(earthTopoTf_url)
+    earthStnSpk = get_path_from_url(earthStnSpk_url)
+    earthHighPerPck = get_path_from_url(earthHighPerPck_url)
+    cklpfkrn = get_path_from_url(cklpfkrn_url)
+# =============================================================================
+#     phobosDsk = get_path_from_url(phobosDsk_url)
+#     marsSpk = get_path_from_url(marsSpk_url)
+#     mroFk = get_path_from_url(mroFk_url)
+#     geophKer = get_path_from_url(geophysical_url)
+#     mro2007sub = get_path_from_url(mro2007sub_url)
+#     spk430sub = get_path_from_url(spk430sub_url)
+# =============================================================================
+    vexboomck = get_path_from_url(vexboomck_url)
+    v02swuck = get_path_from_url(v02swuck_url)    
+    
+def cleanup_file(path):
+    if os.path.exists(path):
+        os.remove(path)
+    pass
+    
+
+def get_kernel(url):
+    kernel_name = get_kernel_name_from_url(url)
+    kernel_file = os.path.join(server, kernel_name)
```

Comment:
You're joining `server`, which is a URL, to a filename, and then seeing if it exists on your local disk. It never will, because you don't have a file named something like `https://pds-rings...`. Maybe you want to use `cwd` here instead?